gulpfile.js

```diff
@@ -1,11 +1,85 @@
-var gulp = require('gulp');
-var uglify = require('gulp-uglify');
-var rename = require('gulp-rename');
 
-gulp.task('build', function() {
- gulp.src('diver.js')
-     .pipe(gulp.dest('dist/'))
-     .pipe(rename('diver.min.js')) 
-     .pipe(uglify())
-     .pipe(gulp.dest('dist'))
+var
+    gulp = require('gulp'),
+    uglify = require('gulp-uglify'),
+    rename = require('gulp-rename'),
+    header = require('gulp-header'),
+    clean = require('gulp-clean'),
+    connect = require('gulp-connect'),
+    rename = require('gulp-rename'),
+    uglify = require('gulp-uglify'),
+    sourcemaps = require('gulp-sourcemaps'),
+    sequence = require('gulp-sequence'),
+    pkg = require('./package.json');
+
+var port = 3000,
+    banner = ['/*!',
+        ' * <%= pkg.name %> - <%= pkg.description %>',
+        ' * <%= pkg.homepage %>',
+        ' * Version: <%= pkg.version %>',
+        ' *',
+        ' * Copyright 2016',
+        ' * Released under the <%= pkg.license %> license',
+        ' */',
+        ''
+    ].join('\n');
+
+
+// cleans the previous build output
+gulp.task('clean', () => {
+    return gulp.src('./dist/*', {
+        read: false
+    })
+        .pipe(clean());
+});
+
+// compiles the source code into '/dist/'
+
+gulp.task('js', () => {
+    return gulp.src('src/diver.js')
+        .pipe(header(banner, {
+            pkg: pkg
+        }))
+        .pipe(gulp.dest('./dist'))
+        .pipe(gulp.dest('./examples/'));
```

Comment:
diver.js shouldn't be inside the examples directory. It can be placed inside lib (a sub dir inside the examples dir)